core/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -39,6 +39,7 @@
         private final ContentPath contentPath;
 
         public BuilderContext(Settings indexSettings, ContentPath contentPath) {
+            assert indexSettings != null;
```

Comment:
I'd probably use `checkNotNull(indexSettings, "indexSettings is required")` here instead. Usually I don't think it is needed but since this used to be nullable it'll be nice to have the assertion in production code.